app/org/sagebionetworks/bridge/services/AuthenticationService.java

```diff
@@ -238,6 +254,40 @@ public void resetPassword(PasswordReset passwordReset) throws BridgeServiceExcep
         accountDao.resetPassword(passwordReset);
     }
     
+    private void repairConsents(Study study, Account account, UserSession session, ClientInfo clientInfo){
+        boolean repaired = false;
+        
+        Map<SubpopulationGuid,ConsentStatus> statuses = session.getUser().getConsentStatuses();
+        for (Map.Entry<SubpopulationGuid,ConsentStatus> entry : statuses.entrySet()) {
+            ConsentSignature activeSignature = account.getActiveConsentSignature(entry.getKey());
+            ConsentStatus status = entry.getValue();
+
+            if (activeSignature != null && !status.isConsented()) {
+                repairConsent(session, entry.getKey(), activeSignature);
+                repaired = true;
+            }
+        }
+        
+        // These are incorrect since they are based on looking up DDB records, so re-create them.
+        if (repaired) {
+            updateConsentStatuses(study, session.getUser(), clientInfo);
+        }
+    }
+    
+    /**
+     * If the signature exists but consentStatus says the user is not consented, then the record is missing in
+     * DDB. We need to create it.
+     */
+    private void repairConsent(UserSession session, SubpopulationGuid subpopGuid, ConsentSignature activeSignature) {
+        logger.error("Signature found without a matching user consent record. Adding consent for " + session.getUser().getId());
+        long signedOn = activeSignature.getSignedOn();
+        if (signedOn == 0L) {
+            signedOn = DateTimeUtils.currentTimeMillis(); // this is so old we did not record a signing date...
+        }
```

Comment:
Is it possible to test this specific case? Evidence suggests that all affected users do not have a signedOn in their Stormpath consent sig. Ergo, this is the most important case to test.